compose/service.py

```diff
@@ -619,3 +623,32 @@ def split_port(port):
 
     external_ip, external_port, internal_port = parts
     return internal_port, (external_ip, external_port or None)
+
+
+def build_extra_hosts(extra_hosts_config):
+    if not extra_hosts_config:
+        return {}
+
+    if isinstance(extra_hosts_config, list):
+        extra_hosts_dict = {}
+        for extra_hosts_line in extra_hosts_config:
+            if isinstance(extra_hosts_line, dict):
```

Comment:
Where the code is updated ?
Did you fix the unit tests i wrote which rely with dict format ?

2015-04-14 16:01 GMT+02:00 chernjie <notifications@github.com>:

> In compose/service.py
> <https://github.com/docker/compose/pull/1158#discussion_r28328697>:
>
> > @@ -619,3 +623,32 @@ def split_port(port):
> >
> >      external_ip, external_port, internal_port = parts
> >      return internal_port, (external_ip, external_port or None)
> > +
> > +
> > +def build_extra_hosts(extra_hosts_config):
> > +    if not extra_hosts_config:
> > +        return {}
> > +
> > +    if isinstance(extra_hosts_config, list):
> > +        extra_hosts_dict = {}
> > +        for extra_hosts_line in extra_hosts_config:
> > +            if isinstance(extra_hosts_line, dict):
>
> Updated code
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/compose/pull/1158/files#r28328697>.
>



-- 
Thomas Desvenain

Téléphone : 09 51 37 35 18